js/modules/k6/http/request.go

```diff
@@ -120,19 +115,28 @@ func (h *HTTP) Request(ctx context.Context, method string, url goja.Value, args
 //TODO break this function up
 //nolint: gocyclo
 func (h *HTTP) parseRequest(
-	ctx context.Context, method string, reqURL httpext.URL, body interface{}, params goja.Value,
+	ctx context.Context, method string, reqURL goja.Value, body interface{}, params goja.Value,
 ) (*httpext.ParsedHTTPRequest, error) {
 	rt := common.GetRuntime(ctx)
 	state := lib.GetState(ctx)
 	if state == nil {
 		return nil, ErrHTTPForbiddenInInitContext
 	}
 
+	u, err := httpext.ToURL(reqURL.Export())
+	if err != nil {
+		if state.Options.Throw.Bool {
+			return nil, err
+		}
+		state.Logger.WithField("error", err).Warn("Request Failed")
+		return nil, nil
```

Comment:
This `nil, nil` return is sketchy, but it would need defining another error for the `http` module since we can't check for the specific `K6Error` error code, as they're not exported. So I'd prefer fixing this later after the error codes refactor.

Reply:
The problem is that now 
```
var resp = http.get("http:// my bad url.com")
resp.status
```
Will be an exception of the type `TypeError: Cannot read property 'status' of undefined` as will anything else touching the response. 
This also will be the first time we actually have `null` returned for the response when we don't actually also throw an exception (return error).

SO at least return an "empty" Response ?

Reply:
Yeah, that would be a breaking change. Though like I mentioned [here](https://github.com/k6io/k6/pull/2045#discussion_r642532999), this makes more sense to me since there was no request made, regardless of the value of `throw`. Which is not the same as if we actually made the request and should return some response.

But yeah, you're probably right that an empty `Response` as done in #1853 would be safer.

Reply:
Resolved by f2665be0.